### PR TITLE
update

### PR DESCRIPTION
**Pull request by paipeng**

### CODE REVIEW

**Comment by paipeng** [comment]
update

File: src/Qt-RSA/qrsaencryption.h [hunk 1]
```diff
@@ -53,7 +53,8 @@ To ensure reliable protection, it is recommended to use an exponent size of at l
         /// Fast but not stable. (using by default)
         Auto = 0,
         /// Stable but slow. (using for sig and check sig messages)
-        OneByte = 1
+        OneByte = 1,
+        Auto2 = 2
```
Comment (EndrII) [inline src/Qt-RSA/qrsaencryption.h:57]: This is bad name of mode. Please add a readable description for this mode and rename it.

File: src/Qt-RSA/qrsaencryption.cpp [hunk 2]
```diff
@@ -315,6 +316,8 @@ QByteArray QRSAEncryption::encode(const QByteArray &rawData, const QByteArray &p
 
     if (blockSizeMode == BlockSize::Auto) {
         blockSizeIn = getBlockSize(m);
+    } else if (blockSizeMode == BlockSize::Auto2) {
+        blockSizeIn = getBlockSize(m) + 1;
```
Comment (EndrII) [inline src/Qt-RSA/qrsaencryption.cpp:320]: For what you use signing block with oversize ?

File: src/Qt-RSA/qrsaencryption.cpp [hunk 3]
```diff
@@ -393,12 +398,47 @@ bool QRSAEncryption::checkSignMessage(const QByteArray &rawData, const QByteArra
...
 
     // if recievedHash == hashAlgorithm(recived message), then signed message is valid
     return recievedHash == QCryptographicHash::hash(message, HashAlgorithm::Sha256);
 }
 
+QByteArray QRSAEncryption::signMessageJava(QByteArray rawData, const QByteArray &privKey, const BlockSize blockSizeMode) {
+    QByteArray hash = QCryptographicHash::hash(rawData, HashAlgorithm::Sha256);
+
+    // ISO Object Identifier
+    QString digestOIDHexString ="3031300d060960864801650304020105000420";
```
Comment (EndrII) [inline src/Qt-RSA/qrsaencryption.cpp:411]: Please add full description about this magic number ...

File: src/Qt-RSA/qrsaencryption.cpp [hunk 4]
```diff
@@ -393,12 +398,47 @@ bool QRSAEncryption::checkSignMessage(const QByteArray &rawData, const QByteArra
...
+    QByteArray digestOID = QByteArray::fromHex(digestOIDHexString.toUtf8());
+    qDebug() << "digestOID: " << QString(digestOID.toHex());
+
+    // Padded
+    QString paddedHexString ="0001ffffffffffffffffffff00";
```
Comment (EndrII) [inline src/Qt-RSA/qrsaencryption.cpp:416]: add description

File: src/Qt-RSA/qrsaencryption.cpp [hunk 5]
```diff
@@ -393,12 +398,47 @@ bool QRSAEncryption::checkSignMessage(const QByteArray &rawData, const QByteArra
...
+    digestOID.append(hash);
+    padded.append(digestOID);
+    qDebug() << "encode: " << QString(padded.toHex());
+
+    QByteArray signature = encode(padded, privKey, blockSizeMode);
+    qDebug() << "signature: " << QString(signature.toHex());
+
+    //rawData.append(QString(SIGN_MARKER + signature.toHex() + SIGN_MARKER).toLatin1());
+
+    return signature;
```
Comment (EndrII) [inline src/Qt-RSA/qrsaencryption.cpp:429]: I still no understand what you sign? 
you have a raw data that want to sign, but you sign them with magic arrays. 
please attach full description of this function.

File: src/Qt-RSA/qrsaencryption.cpp [hunk 6]
```diff
@@ -393,12 +398,47 @@ bool QRSAEncryption::checkSignMessage(const QByteArray &rawData, const QByteArra
...
+}
+bool QRSAEncryption::checkSignMessageJava(const QByteArray &message, const QByteArray &rawData, const QByteArray &pubKey, const BlockSize blockSizeMode) {
+    // start position of SIGN_MARKER in rawData
+    auto signStartPos = 19+ 13;
```
Comment (EndrII) [inline src/Qt-RSA/qrsaencryption.cpp:433]: Why sign raker began of this 19 + 13.
Please describe your code. 

If you want to found a beginning of your sign, use hooks (or magic symbol, you need to find it on array before processing).

Or use constant variables with readable names.

File: src/Qt-RSA/qrsaencryption.cpp [hunk 7]
```diff
@@ -431,6 +471,26 @@ QByteArray QRSAEncryption::load(const QString &file) {
     return {};
 }
 
+
+QByteArray QRSAEncryption::loadBinary(const QString &file) {
+    QFile keyFile(file);
+
+    if (keyFile.open(QIODevice::ReadOnly)) {
+#if 0
```
Comment (EndrII) [inline src/Qt-RSA/qrsaencryption.cpp:479]: Or remove it or add useful condition.

File: src/Qt-RSA/qrsaencryption.h [hunk 8]
```diff
@@ -216,17 +219,22 @@ To ensure reliable protection, it is recommended to use an exponent size of at l
...
      * @param rawData The raw data for verification.
      * @param pubKey The public key.
+     * @param blockSizeMode block Size.
      * @return true if data sigend.
      */
-    bool checkSignMessage(const QByteArray &rawData, const QByteArray &pubKey);
+    bool checkSignMessage(const QByteArray &rawData, const QByteArray &pubKey, const BlockSize blockSizeMode = BlockSize::OneByte);
+
+    QByteArray signMessageJava(QByteArray rawData, const QByteArray &privKey, const BlockSize blockSizeMode = BlockSize::OneByte);
+    bool checkSignMessageJava(const QByteArray &message, const QByteArray &rawData, const QByteArray &pubKey, const BlockSize blockSizeMode = BlockSize::OneByte);
```
Comment (EndrII) [inline src/Qt-RSA/qrsaencryption.h:237]: Add XML description for all new methods

**Review by EndrII — CHANGES_REQUESTED** [review]
Thank you for your work. But these changes need to be improved.

**Comment by EndrII** [comment]
so rename please pull request.

**Comment by EndrII** [comment]
Unfortunately. But I drop support of this library. See my new library for works with RSA and ECDSA.  [easyssl](https://github.com/QuasarApp/easyssl)

**Pull request closed without merging**